files/common/usr/bin/unpack-image

```diff
@@ -103,14 +104,34 @@ for file in SHA256SUMS prepare; do
 	[[ -f "$file" ]] || die "image is corrupt; missing '$file' file"
 done
 
+
+#
+# Provides the version of the latest public key in this engine.
+# Returns "6.0" if the latest key was made at version 6.0 or earlier.
+#
+function get_latest_public_key_version
+{
+	local public_key_dir=$1
+
+	# Public keys are named key-public.pem[.type.version] (for versions above 5.2)
+	(set -o pipefail; ls $public_key_dir/key-public.pem* \
```

Comment:
Can you please describe each step of this pipeline? As is, it's hard for me to determine exactly what each step is doing, and why it's doing what it's doing..

This code only has to handle version 6.0 and later (right?), so I'm concerned that if we simply copy what the app-gate code was doing, we may inherit additional complexity that is not needed here.

Further, I believe we'll only ever have a single key installed in `/var/lib/delphix-appliance` at any given time (right?), so I'm wondering if we can more simply just use:
```
$ ls /var/lib/delphix-appliance/key-public.pem.upgrade.*
/var/lib/delphix-appliance/key-public.pem.upgrade.5.3
```
?

Reply:
Will get back to you offline on this.
If its handling only version 6.0 and later, agree with you on simplifying it per your suggestion above.

Reply:
Fixed.